src/frontend/src/lib/utils/token-group.utils.ts

```diff
@@ -68,19 +64,60 @@ export const groupTokensByTwin = (tokens: TokenUi[]): TokenUiOrGroupUi[] => {
 			return token;
 		}
 
-		const twinToken = tokens.find((t) => t.symbol === token.twinTokenSymbol && isIcCkToken(t)) as
-			| IcCkToken
-			| undefined;
+		if (groupedTokenTwins.has(token.symbol)) {
+			return token;
+		}
+
+		const relatedTokens: TokenUi[] = [];
+
+		// Find ckToken twin
+		const ckTwinToken = tokens.find(
+			(t) => t.symbol === token.twinTokenSymbol && isIcCkToken(t) && t.decimals === token.decimals
+		) as IcCkToken | undefined;
+
+		if (ckTwinToken && !groupedTokenTwins.has(ckTwinToken.symbol)) {
+			relatedTokens.push({ ...ckTwinToken, enabled: true });
+			groupedTokenTwins.add(ckTwinToken.symbol);
+		}
 
-		if (twinToken && twinToken.decimals === token.decimals) {
-			groupedTokenTwins.add(twinToken.symbol);
+		// Find Bitfinity token twin (starts with 'o')
+		const bitfinityTwinToken = tokens.find(
+			(t) =>
+				(t.symbol === `o${token.symbol}` && t.decimals === token.decimals) ||
```

Comment:
I think it should be this:
```
t.symbol === `o${token.symbol}`
```
or this:
```
t.symbol.startsWith('o')
```
?

maybe put `t.symbol.toLowerCase()` and ``o${token.symbol}`.toLowerCase()`